src/ir/AST/Util.hs

```diff
@@ -475,8 +487,13 @@ mark asParent s@Let{body, decls} =
   where
     markDecl (n, e) = (n, markAsExpr e)
 mark asParent s@While{cond, body} = asParent s{cond=markAsExpr cond, body=markAsStat body}
-mark asParent s@For{step, src, body} =
-  asParent s{step=markAsExpr step, src=markAsExpr src, body=markAsStat body}
+-- JOY for-comprehension, doesn't work, type matching is wrong.
+mark asParent s@For{sources, body} =
+  asParent s{sources = map markAsForSource sources, body=markAsStat body}
+  where
+    markAsForSource ForSource{forVar, forVarType, collection} = ForSource {forVar, forVarType, collection = markAsExpr collection}
+{-mark asParent s@For{step, src, body} =
```

Comment:
remove this

Reply:
try to stick to the 80 characters width rule, I think this is mostly used in the whole code base